src/contrasts.jl

```diff
@@ -83,12 +90,37 @@ type ContrastsMatrix{C <: AbstractContrasts, T}
 end
 
 """
-    ContrastsMatrix{C<:AbstractContrasts}(contrasts::C, levels::AbstractVector)
+An instantiation of a contrast coding strategy for particular levels
 
-Compute contrasts matrix for given data levels.
+This type is used internally for generating model matrices based on categorical
+data, and __most users will not need to deal with it directly__.  It's exported
+for developers that need to extend these facilities.  Conceptually, a
```

Comment:
Does it have to be exported in order to be extended by another package? If not, we should probably just not export it since it isn't intended to be user-facing.